src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPGlobalStatistics.cs

```diff
@@ -0,0 +1,249 @@
+            string fileContents = File.ReadAllText(LinuxNetworkFiles.SnmpV4StatsFile);
+            int firstIpHeader = fileContents.IndexOf("Ip:");
+            int secondIpHeader = fileContents.IndexOf("Ip:", firstIpHeader + 1);
+            int endOfSecondLine = fileContents.IndexOf(Environment.NewLine, secondIpHeader);
+            string ipData = fileContents.Substring(secondIpHeader, endOfSecondLine - secondIpHeader);
+            StringParser parser = new StringParser(ipData, ' ');
+
+            // NOTE: Need to verify that this order is consistent. Otherwise, we need to parse the first-line header
+            // to determine the order of information contained in the file.
+
+            parser.MoveNextOrFail(); // Skip Ip:
+
+            // According to RFC 1213, "1" indicates "acting as a gateway". "2" indicates "NOT acting as a gateway".
+            stats._forwarding = parser.MoveAndExtractNext() == "1";
+            stats._defaultTtl = parser.ParseNextInt32();
+            stats._inReceives = parser.ParseNextInt32();
+            stats._inHeaderErrors = parser.ParseNextInt32();
+            stats._inAddressErrors = parser.ParseNextInt32();
+            stats._forwardedDatagrams = parser.ParseNextInt32();
+            stats._inUnknownProtocols = parser.ParseNextInt32();
+            stats._inDiscards = parser.ParseNextInt32();
+            stats._inDelivers = parser.ParseNextInt32();
+            stats._outRequests = parser.ParseNextInt32();
+            stats._outDiscards = parser.ParseNextInt32();
+            stats._outNoRoutes = parser.ParseNextInt32();
+            stats._reassemblyTimeout = parser.ParseNextInt32();
+            stats._reassemblyRequireds = parser.ParseNextInt32();
+            stats._reassemblyOKs = parser.ParseNextInt32();
+            stats._reassemblyFails = parser.ParseNextInt32();
+            stats._fragmentOKs = parser.ParseNextInt32();
+            stats._fragmentFails = parser.ParseNextInt32();
+            stats._fragmentCreates = parser.ParseNextInt32();
+
+            // Experimental
+            string routeFile = ReadProcConfigFile(LinuxNetworkFiles.Ipv4RouteFile);
+            stats._numRoutes = CountOccurences(Environment.NewLine, routeFile) - 1; // File includes one-line header
+
+            // Just count the number of files under /proc/sys/net/ipv4/conf,
+            // because GetAllNetworkInterfaces() is relatively expensive just for the count.
+            int interfacesCount = 0;
+            var files = new DirectoryInfo(LinuxNetworkFiles.Ipv4ConfigFolder).GetFiles();
+            foreach (var file in files)
+            {
+                if (file.Name != "all" && file.Name != "default")
+                {
+                    interfacesCount++;
+                }
+            }
+            stats._numInterfaces = interfacesCount;
+
+            return stats;
+        }
+
+        /// <summary>
+        /// Constructs an IPGlobalStatistics object from the snmp6 stats file.
```

Comment:
These are the names of the files on disk ("/proc/net/snmp", "/proc/net/snmp6"), so I am inclined to leave it, even though it's an acronym